File: sources/go/test/Test_LinkList.go
package test

import (
	"fmt"
	"structure/list/linklist"
)

func TestLinkList() {

	fmt.Printf("********** 单链表测试开始 **********\n")

	l := linklist.NewLinkList()

	fmt.Println("追加元素-------------")
	l.Append(17)
	l.Append(20)
	l.Display()

	fmt.Println("插入元素-------------")
	l.Insert(2, 18)
	l.Insert(3, 19)
	l.Display()

	fmt.Println("修改元素-------------")
	l.Update(2, 22)
	l.Display()

	fmt.Println("查找元素-------------")
	te, _ := l.LocateElem(19)
	pe, _ := l.PriorElem(22)
	ne, _ := l.NextElem(22)
	fmt.Printf("查找第三个元素位置：%d\n", te)
	fmt.Printf("查找第二个元素前驱：%d\n", pe)
	fmt.Printf("查找第二个元素后继：%d\n", ne)

	fmt.Println("删除元素-------------")
	l.Delete(2)
	l.Display()

	fmt.Printf("********** 测试结束 **********\n")

}

File: sources/js/test/Test_LinkList.js
import LinkList from "../list/LinkList";

function TestLinkList(){

    console.log("********** 顺序表测试开始 **********");

    let l = new LinkList();
    l.display()

    console.log("增加元素-------------");
    l.append(17)
    l.append(20)
    l.display()

    console.log("插入元素-------------");
    l.insert(2, 18)
    l.insert(3, 19)
    l.display()

    console.log("修改元素-------------");
    l.update(2, 22)
    l.display()

    console.log("查找第二个元素前驱：", l.priorElem(22));
    console.log("查找第二个元素后继：", l.nextElem(22));

    console.log("删除元素-------------");
    l.delete(2)
    l.display()

    console.log("********** 顺序表测试结束 **********");
}

export default TestLinkList